starknet_py/contract.py

```diff
@@ -701,35 +942,112 @@ async def deploy_contract(
         nonce: Optional[int] = None,
         max_fee: Optional[int] = None,
         auto_estimate: bool = False,
+        salt: Optional[int] = None,
+        unique: bool = True,
     ) -> "DeployResult":
         """
-        Deploys a contract through Universal Deployer Contract
+        Deploys a contract through Universal Deployer Contract.
 
         :param account: BaseAccount used to sign and send deploy transaction.
         :param class_hash: The class_hash of the contract to be deployed.
         :param abi: An abi of the contract to be deployed.
         :param constructor_args: a ``list`` or ``dict`` of arguments for the constructor.
         :param deployer_address: Address of the UDC. Is set to the address of
-            the default UDC (same address on mainnet/testnet/devnet) by default.
+            the default UDC (same address on mainnet/goerli/sepolia) by default.
             Must be set when using custom network other than ones listed above.
         :param cairo_version: Version of the Cairo in which contract is written.
+            By default, it is set to 0.
```

Comment:
Nit: I would consider removing default there, as we're slowly moving away from cairo 0.

Alternatively, and that's just a rough idea for the future, we could check the version of declare transaction used to declare contract at given class hash and determine cairo version that way.

Reply:
Are you planning to add an issue for this?

Reply:
https://github.com/software-mansion/starknet.py/issues/1271